src/test/ui/asm/sym.rs

```diff
@@ -27,12 +28,48 @@ macro_rules! call {
                 out("xmm8") _, out("xmm9") _, out("xmm10") _, out("xmm11") _,
                 out("xmm12") _, out("xmm13") _, out("xmm14") _, out("xmm15") _,
             );
+            result
         }
-        result
-    }}
+    }
 }
 
+macro_rules! static_addr {
+    ($s:expr) => {
+        unsafe {
+            let result: *const u32;
+            asm!("lea {}, [rip + {}]", out(reg) result, sym $s);
```

Comment:
I have absolutely no clue what this or the tls asm below does. Can you document this in source with some comments for each asm instruction and its arguments?

Reply:
I've added some comments on the asm.